spotlight-client/src/contentModels/RacialDisparitiesNarrative.ts

```diff
@@ -0,0 +1,605 @@
+      ...mapping,
+      [currentKey.replace("Count", "Proportion")]:
+        record[currentKey] / totalRevocations,
+    }),
+    {} as RevocationProportions
+  );
+};
+
+type RecordsMapping = { [key in RaceIdentifier]: RacialDisparitiesRecord };
+
+type SentencingMetrics = {
+  incarcerationPctCurrent: number;
+  probationPctCurrent: number;
+};
+
+type SupervisionType = "parole" | "probation" | "supervision";
+
+function getSentencingMetrics(
+  record: RacialDisparitiesRecord
+): SentencingMetrics {
+  const totalSentencedCurrent = record.currentTotalSentencedCount;
+
+  const incarcerationPctCurrent =
+    (record.currentIncarcerationSentenceCount +
+      record.currentDualSentenceCount) /
+    totalSentencedCurrent;
+
+  const probationPctCurrent =
+    (record.currentProbationSentenceCount + record.currentDualSentenceCount) /
+    totalSentencedCurrent;
+
+  return {
+    incarcerationPctCurrent,
+    probationPctCurrent,
+  };
+}
+
+type ConstructorOpts = {
+  tenantId: TenantId;
+  defaultCategory?: RaceIdentifier;
+  defaultSupervisionType?: SupervisionType;
+  content: RacialDisparitiesNarrativeContent;
+};
+
+/**
+ * Data model for the racial disparities narrative, which handles computation
+ * as well as fetching. The preferred method for instantiating this class is
+ * to use the static `build` method.
+ * @example
+ *
+ * ```js
+ * const narrative = RacialDisparitiesNarrative.build(props);
+ * ```
+ */
+export default class RacialDisparitiesNarrative {
+  // metadata
+  readonly title = "Racial Disparities";
+
+  readonly chartLabels: RacialDisparitiesChartLabels;
+
+  readonly tenantId: TenantId;
+
+  private readonly focusColor = colors.dataViz[0];
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  private readonly unfocusedColor = colors.dataVizNamed.get("paleBlue")!;
+
+  // API data
+  isLoading?: boolean;
+
+  private records?: RecordsMapping;
+
+  error?: Error;
+
+  // filters
+  selectedCategory: RaceIdentifier;
+
+  supervisionType: SupervisionType;
+
+  static build(props: ConstructorOpts): RacialDisparitiesNarrative {
+    return new RacialDisparitiesNarrative(props);
+  }
+
+  constructor({
+    tenantId,
+    defaultCategory,
+    defaultSupervisionType,
+    content,
+  }: ConstructorOpts) {
+    this.tenantId = tenantId;
+    this.selectedCategory = defaultCategory || "BLACK";
+    this.supervisionType = defaultSupervisionType || "supervision";
+    this.chartLabels = content.chartLabels;
+
+    makeAutoObservable<RacialDisparitiesNarrative, "records">(this, {
+      records: observable.ref,
+    });
+  }
+
+  /**
+   * Fetches metric data and stores the result reactively on this Metric instance.
+   */
+  async hydrate(): Promise<void> {
+    this.isLoading = true;
+    try {
+      const fetchedData = await fetchAndTransformMetric({
+        sourceFileName: "racial_disparities",
+        tenantId: this.tenantId,
+        transformFn: createRacialDisparitiesRecords,
```

Comment:
This is a really nice pattern. Now, having seen the limitations of our DataStores, I appreciate more this approach.